pallets/funding/src/tests/1_application.rs

```diff
@@ -0,0 +1,1321 @@
+			inst.execute(|| {
+				find_event!(TestRuntime, Event::<TestRuntime>::MetadataEdited{project_id, ref metadata}, project_id == 0, metadata == &new_metadata_1);
+			});
+
+			// All fields changed
+			assert_ok!(inst.execute(|| crate::Pallet::<TestRuntime>::edit_project(
+				RuntimeOrigin::signed(ISSUER_1),
+				jwt.clone(),
+				project_id,
+				new_metadata_2.clone()
+			)));
+			assert_eq!(inst.get_project_metadata(project_id), new_metadata_2);
+			inst.execute(|| {
+				find_event!(TestRuntime, Event::<TestRuntime>::MetadataEdited{project_id, ref metadata}, project_id == 0, metadata == &new_metadata_2);
+			});
+		}
+
+		#[test]
+		fn adding_offchain_hash() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let mut project_metadata = default_project_metadata(inst.get_new_nonce(), ISSUER_1);
+			project_metadata.offchain_information_hash = None;
+			inst.mint_plmc_to(default_plmc_balances());
+			let jwt = get_mock_jwt(ISSUER_1, InvestorType::Institutional, generate_did_from_account(ISSUER_1));
+			let project_id = inst.create_new_project(project_metadata.clone(), ISSUER_1);
+			let mut new_metadata = project_metadata.clone();
+			new_metadata.offchain_information_hash = Some(hashed(METADATA));
+			assert_ok!(inst.execute(|| crate::Pallet::<TestRuntime>::edit_project(
+				RuntimeOrigin::signed(ISSUER_1),
+				jwt.clone(),
+				project_id,
+				new_metadata.clone()
+			)));
+			assert_eq!(inst.get_project_metadata(project_id), new_metadata);
+		}
+
+		#[test]
+		fn storage_changes() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let project_metadata = default_project_metadata(inst.get_new_nonce(), ISSUER_1);
+			inst.mint_plmc_to(default_plmc_balances());
+			let jwt = get_mock_jwt(ISSUER_1, InvestorType::Institutional, generate_did_from_account(ISSUER_1));
+			let project_id = inst.create_new_project(project_metadata.clone(), ISSUER_1);
+			let mut new_metadata = project_metadata.clone();
+			new_metadata.total_allocation_size = 100_000 * ASSET_UNIT;
+			new_metadata.minimum_price = PriceOf::<TestRuntime>::from_float(1f64);
+			assert_ok!(inst.execute(|| crate::Pallet::<TestRuntime>::edit_project(
+				RuntimeOrigin::signed(ISSUER_1),
+				jwt.clone(),
+				project_id,
+				new_metadata.clone()
+			)));
+			// Project details reflect changes
+			assert_eq!(inst.get_project_details(project_id).fundraising_target, 100_000 * US_DOLLAR);
+			// Bucket reflects changes
+			let new_bucket = Pallet::<TestRuntime>::create_bucket_from_metadata(&new_metadata).unwrap();
+			let stored_bucket = inst.execute(|| Buckets::<TestRuntime>::get(project_id).unwrap());
+			assert_eq!(stored_bucket, new_bucket);
+			// Event emitted
+			inst.execute(|| {
+				find_event!(TestRuntime, Event::<TestRuntime>::MetadataEdited{project_id, ref metadata}, project_id == 0, metadata == &new_metadata);
+			});
+		}
+	}
+
+	#[cfg(test)]
+	mod failure {
```

Comment:
We should probably remove all duplicate tests that check certain logic in the project_validation function as its the exact same logic being used in create_project. So only do tests here that are actually relevant to the lines that are not covered by create_project. So:
Issuer and project_frozen

Reply:
Or like Albert Einstein said: "Insanity is doing the same thing over and over again and expecting a different result" 😄

Reply:
Like mentioned before, the reasoning here was decoupling the intended behavior from the tech implementation. If @lrazovic also thinks we should remove it and trust we will use the same implementation on both logic paths then let's do it

Reply:
I would be on the side of testing the "tech implementation" here and the "intended behavior" in the various integration tests via xcm-emulator and/or chopsticks